### PR TITLE
Use pyplot.get_cmap for compatiblity with matplotlib 3.3 to 3.6 in in show_rag

### LINKED ISSUE #6486: Address new matplotlib >= 3.5 dependency in skimage.future.graph.rag.show_rag
## Description

With #6483, `skimage.future.graph.rag.show_rag` now requires matplotlib >= 3.5 which trips up our CI when matplotlib 3.4 or earlier are used. We can either (a) bump our matplotlib for the whole package (currently >=3.3) or (b) we configure our test suite to ignore it "for now". I'm voting for the first (a).

## Way to reproduce

Run doctests with matplotlib < 3.5.

### PR DESCRIPTION
**Pull request by lagru**
## Description

Updates the solution from 514c08923cbbdbcb6f8e8165806f4728a3ad7112 (#6483) to one that is compatible with all currently supported matplotlib versions and will work with 3.6 also. This allows us to not update our matplotlib dependency to 3.5 yet, which might be a bit early.

Marking as draft because I want to get this merged before the CI is green. ;)

Closes #6486

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Review by alexdesiqueira — APPROVED** [review]

**Comment by alexdesiqueira** [comment]
Thank you @lagru! CI :green_circle:; feel free to merge :)

File: skimage/future/graph/rag.py [hunk 1]
```diff
@@ -498,7 +498,7 @@ def show_rag(labels, rag, image, border_color='black', edge_width=1.5,
     >>> lc = graph.show_rag(labels, g, img)
     >>> cbar = plt.colorbar(lc)
     """
-    from matplotlib import colors, colormaps
+    from matplotlib import colors, pyplot
```
Comment (rfezzani) [inline skimage/future/graph/rag.py:501]:
```suggestion
    from matplotlib import colors
```
`pyplot` is already imported as `plt` in the line below :wink:

File: skimage/future/graph/rag.py [hunk 2]
```diff
@@ -516,12 +516,12 @@ def show_rag(labels, rag, image, border_color='black', edge_width=1.5,
         # Ignore the alpha channel
         out = image[:, :, :3]
     else:
-        img_cmap = colormaps[img_cmap]
+        img_cmap = pyplot.get_cmap(img_cmap)
```
Comment (rfezzani) [inline skimage/future/graph/rag.py:519]:
```suggestion
        img_cmap = plt.get_cmap(img_cmap)
```

File: skimage/future/graph/rag.py [hunk 3]
```diff
@@ -516,12 +516,12 @@ def show_rag(labels, rag, image, border_color='black', edge_width=1.5,
...
         out = color.rgb2gray(image)
         # Ignore the alpha channel
         out = img_cmap(out)[:, :, :3]
 
-    edge_cmap = colormaps[edge_cmap]
+    edge_cmap = pyplot.get_cmap(edge_cmap)
```
Comment (rfezzani) [inline skimage/future/graph/rag.py:524]:
```suggestion
    edge_cmap = plt.get_cmap(edge_cmap)
```

**Review by rfezzani — APPROVED** [review]
Thank you @lagru, just suggesting to remove a duplicate import of `pyplot`

File: skimage/future/graph/rag.py (hunk 1 above)
Comment (lagru) [reply skimage/future/graph/rag.py:501]: Ha, yes. Good catch! :sweat_smile:

**Comment by rfezzani** [comment]
Thank you @lagru

**Pull request merged**